sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/AnalyzeSentimentBatchDocumentsAsync.java

```diff
@@ -0,0 +1,97 @@
+import com.azure.ai.textanalytics.models.TextDocumentBatchStatistics;
+import com.azure.ai.textanalytics.models.TextDocumentInput;
+import com.azure.ai.textanalytics.models.TextSentiment;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample demonstrate how to analyze sentiment of a batch of text inputs in asynchronously call.
+ */
+public class AnalyzeSentimentBatchDocumentsAsync {
+    /**
+     * Main method to invoke this demo about how to analyze sentiment of a batch of text inputs.
+     *
+     * @param args Unused arguments to the program.
+     */
+    public static void main(String[] args) {
+        // Instantiate a client that will be used to call the service.
+        TextAnalyticsAsyncClient client = new TextAnalyticsClientBuilder()
+            .subscriptionKey("<replace-with-your-text-analytics-key-here>")
+            .endpoint("<replace-with-your-text-analytics-endpoint-here>")
+            .buildAsyncClient();
+
+        // The texts that need be analysed.
+        List<TextDocumentInput> inputs = Arrays.asList(
+            new TextDocumentInput("1", "The hotel was dark and unclean. The restaurant had amazing gnocchi.", "en"),
+            new TextDocumentInput("2", "The restaurant had amazing gnocchi. The hotel was dark and unclean.", "en")
+        );
+
+        // Request options: show statistics and model version
+        final TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions().setShowStatistics(true);
+
+        // Analyzing batch sentiments
+        client.analyzeBatchSentimentWithResponse(inputs, requestOptions).subscribe(
+            result -> {
+                DocumentResultCollection<AnalyzeSentimentResult> analyzedBatchResult = result.getValue();
+                System.out.printf("Model version: %s%n", analyzedBatchResult.getModelVersion());
+
+                // Batch statistics
+                final TextDocumentBatchStatistics batchStatistics = analyzedBatchResult.getStatistics();
+                System.out.printf("A batch of document statistics, document count: %s, erroneous document count: %s, transaction count: %s, valid document count: %s.%n",
+                    batchStatistics.getDocumentCount(),
+                    batchStatistics.getErroneousDocumentCount(),
+                    batchStatistics.getTransactionCount(),
+                    batchStatistics.getValidDocumentCount());
+
+                // Analyzed sentiment for each of document from a batch of documents
+                for (AnalyzeSentimentResult analyzeSentimentResult : analyzedBatchResult) {
+                    System.out.printf("Document ID: %s%n", analyzeSentimentResult.getId());
+                    final TextSentiment documentSentiment = analyzeSentimentResult.getDocumentSentiment();
+                    // Erroneous document
+                    if (documentSentiment == null) {
```

Comment:
That seems like an odd way to check for error. I thought you had a `hasError()` method?